File: restaurant_pos_web/backend/Models/OrderItem.cs
using System.ComponentModel.DataAnnotations;

namespace RestaurantPos.Api.Models;

public class OrderItem
{
    [Key]
    public int Id { get; set; }
    public string OrderId { get; set; } = string.Empty;
    public string ProductId { get; set; } = string.Empty;
    public string? ProductName { get; set; }
    public decimal UnitPrice { get; set; }
    public int Quantity { get; set; }
    public decimal LineTotal { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

File: restaurant_pos/lib/models/order_item.dart
import 'product.dart';

class OrderItem {
  final int id;
  final String orderId;
  final String productId;
  final String? productName;
  final double unitPrice;
  final int quantity;
  final double lineTotal;
  final String? notes;
  
  // Extra field for UI convenience
  final Product? product;

  OrderItem({
    this.id = 0,
    this.orderId = '',
    required this.productId,
    this.productName,
    required this.unitPrice,
    required this.quantity,
    required this.lineTotal,
    this.notes,
    this.product,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      id: (json['id'] as num?)?.toInt() ?? 0,
      orderId: json['orderId']?.toString() ?? '',
      productId: json['productId']?.toString() ?? '',
      productName: json['productName'],
      unitPrice: (json['unitPrice'] as num?)?.toDouble() ?? (json['price'] as num?)?.toDouble() ?? 0.0,
      quantity: (json['quantity'] as num?)?.toInt() ?? 0,
      lineTotal: (json['lineTotal'] as num?)?.toDouble() ?? 0.0,
      notes: json['notes'],
      product: json['product'] != null ? Product.fromJson(json['product']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'orderId': orderId,
      'productId': productId,
      'productName': productName,
      'unitPrice': unitPrice,
      'quantity': quantity,
      'lineTotal': lineTotal,
      'notes': notes,
      if (product != null) 'product': product!.toJson(),
    };
  }

  // Compatibility getter
  double get price => unitPrice;
}
